src/main/java/com/google/songgame/servlets/GameServlet.java

```diff
@@ -46,21 +50,122 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+    // Retrieve Playlist item from Youtube API
     try {
-      ClientCredentials clientCredentials = clientCredentialsRequest.execute();
-      return clientCredentials.getAccessToken();
-    } catch (IOException | SpotifyWebApiException | ParseException e) {
-      System.out.println("Error: " + e.getMessage());
+      playlistItem = getPlaylistInfo(playlistId);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    // Parse Playlist item Json string to retrieve video IDs
+    String playlistItemJson = new Gson().toJson(playlistItem);
+    ArrayList<String> playlistVideos = parsePlaylistItem(playlistItemJson);
+    playlistSize = playlistVideos.size();
+    videoId = getRandomVideo(playlistVideos);
+  }
+
+  /**
+   * Returns playlist ID
+   */
+  private String getIdFromUrl(String playlistUrl) {
+    if (playlistUrl.contains("youtube.com/playlist?list=")) {
+      int start = playlistUrl.indexOf("list=") + 5;
+      int end = playlistUrl.length();
+      if (playlistUrl.contains("&")) {
+        end = playlistUrl.indexOf("&", start);
+      }
+      return playlistUrl.substring(start, end);
+    } else {
+      System.err.println(playlistUrl + " is not a valid YouTube Playlist URL.");
+      return "";
+      // TODO @hdee: do something more complicated to handle this error.
+      // TODO @hdee: test this method.
     }
-    return "";
   }
 
+  /**
+   * Call function to create API service object.
+   *
+   * @throws GeneralSecurityException, IOException, GoogleJsonResponseException
+   */
+  public static PlaylistItemListResponse getPlaylistInfo(String playlistId)
+      throws GeneralSecurityException, IOException, GoogleJsonResponseException {
+    YouTube youtubeService = getService();
+    // Define and execute the API request
+    YouTube.PlaylistItems.List request =
+        youtubeService
+            .playlistItems()
+            .list("snippet")
+            .setMaxResults(MAX_RESULTS)
+            .setPlaylistId(playlistId);
+    PlaylistItemListResponse response = request.execute();
+    return response;
+  }
+
+  /**
+   * Build and return an authorized API client service.
+   *
+   * Returns an authorized API client service
+   * @throws GeneralSecurityException, IOException
+   */
+  public static YouTube getService() throws GeneralSecurityException, IOException {
+    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    return new YouTube.Builder(httpTransport, GSON_FACTORY, null)
+        .setApplicationName(APPLICATION_NAME)
+        .setYouTubeRequestInitializer(new YouTubeRequestInitializer(DEVELOPER_KEY))
+        .build();
+  }
+
+  /**
+   * Returns an ArrayList of video IDs
+   */
+  private ArrayList<String> parsePlaylistItem(String playlistItemJson) {
+    String[] playlistItemData = playlistItemJson.split("\",\"");
+    ArrayList<String> playlistVideos = new ArrayList<String>();
+    // extract video ID from sections
+    for (String data : playlistItemData)
```

Comment:
First off, it's pretty awesome that you wrote this Json parsing code. I think you can simplify it a bit using the JsonParser class from the Gson library. However, I think this would be a stretch goal, so can you file an issue to do this? You can get to it at a later point if we have time.

Here's an article with more information that you can add to the issue and reference later:
http://tutorials.jenkov.com/java-json/gson-jsonparser.html